Library/include/CSP/Common/List.h

```diff
@@ -277,12 +277,11 @@ template <typename T> class CSP_API List
 	/// @param Item const T& : Element to remove from the list
 	void RemoveItem(const T& Item)
 	{
-		for (auto i = CurrentSize - 1; i >= 0; --i)
+		for (size_t i = 0; i < CurrentSize; ++i)
```

Comment:
CurrentSize is a size_t (uint). So if the value was 0, we were doing (0 - 1), causing the value to be UINT64_MAX. I assume the ordering was originally inverted because we supported removing multiple values, but this isn't the case.